wallet/src/de/schildbach/wallet/ui/dashpay/ContactSearchResultsAdapter.kt

```diff
@@ -78,26 +71,59 @@ class ContactSearchResultsAdapter(private val listener: Listener,
         return results.size
     }
 
+    var sendContactRequestWorkStateMap: Map<String, Resource<WorkInfo>> = mapOf()
+        set(value) {
+            field = value
+            notifyDataSetChanged()
+        }
+
     override fun getItemId(position: Int): Long {
-        return when (results[position].viewType) {
-            CONTACT -> PlatformUtils.longHashFromEncodedString(results[position].usernameSearchResult!!.toContactRequest!!.toUserId)
-            CONTACT_REQUEST -> PlatformUtils.longHashFromEncodedString(results[position].usernameSearchResult!!.fromContactRequest!!.userId)
+        val item = results[position]
+        return when (item.viewType) {
+            CONTACT -> {
+                if (item.usernameSearchResult!!.type == UsernameSearchResult.Type.CONTACT_ESTABLISHED) {
+                    PlatformUtils.longHashFromEncodedString(item.usernameSearchResult.toContactRequest!!.toUserId)
+                } else {
+                    PlatformUtils.longHashFromEncodedString(item.usernameSearchResult.fromContactRequest!!.userId)
+                }
+            }
             CONTACT_REQUEST_HEADER -> 1L
             CONTACT_HEADER -> 2L
-            else -> throw IllegalArgumentException("Invalid viewType ${results[position].viewType}")
+            else -> throw IllegalArgumentException("Invalid viewType ${item.viewType}")
         }
     }
 
-    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        when (results[position].viewType) {
-            CONTACT, CONTACT_REQUEST -> holder.bind(results[position].usernameSearchResult!!)
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        val item = results[position]
+        when (item.viewType) {
+            CONTACT -> {
+                val sendContactRequestWorkState = sendContactRequestWorkStateMap[item.usernameSearchResult!!.dashPayProfile.userId]
+                (holder as ContactViewHolder).apply {
+                    bind(item.usernameSearchResult, sendContactRequestWorkState, itemClickListener, listener)
+                    if (item.usernameSearchResult.isPendingRequest) {
+                        setMarginsDp(20, 3, 20, 3)
+                        setBackgroundResource(R.drawable.selectable_round_corners)
+                    } else {
+                        setMarginsDp(0, 0, 0, 0)
+                        // background color alternates based on first letter
+                        val colorResId = if (item.usernameSearchResult.dashPayProfile.username[0].toLowerCase().toInt() % 2 != 0) {
+                            R.color.white
+                        } else {
+                            R.color.dash_lighter_gray
+                        }
+                        val color = ResourcesCompat.getColor(holder.itemView.resources, colorResId, null)
+                        setBackgroundColor(color)
```

Comment:
This alternating background color code was removed in a merged PR from last week:

https://github.com/dashevo/dash-wallet/pull/514/files

Reply:
Fixed